src/toil/jobStores/utils.py

```diff
@@ -269,8 +287,8 @@ class ReadableTransformingPipe(ReadablePipe):
     
     """
     
-    def __init__(self, source):
-        super(ReadableTransformingPipe, self).__init__()
+    def __init__(self, source, encoding=None, errors=None):
```

Comment:
Previously, the source stream passed here was always assumed to be in binary mode. If you set an encoding on the transforming pipe, should it also expect a text-mode input stream and decode it with that encoding?

If you don't do that, you're going to have trouble swapping a transformation in and out of a pipeline of other text-mode streams; you'd need to ensure all but the last thing in the pipeline are running in binary mode, which I don't think you do for e.g. AWS downloads.